src/app/components/watch-detail/watch-detail.component.ts

```diff
@@ -20,30 +22,47 @@ export class WatchDetailComponent implements OnInit {
 
     public queryParams!: Params;
 
-    private routeSubscription: Subscription;
+    private routeSubscription!: Subscription;
 
     private watchId!: number;
 
+    private destroy$: Subject<void> = new Subject();
+
     constructor(
         private route: ActivatedRoute,
-        private watchesService: WatchesService,
+        private itemsService: WatchesService,
         private cartService: CartService) {
 
-        this.routeSubscription = route.params.subscribe((params: Params) => {
-            this.watchId = parseInt(params['watchId'], 10);
-        }
-        );
+
     }
 
     public ngOnInit(): void {
-        this.watch = this.watchesService.getWatchById(this.watchId);
+        this.routeSubscription = this.route.params.subscribe((params: Params) => {
+            this.watchId = parseInt(params['itemId'], 10);
+        });
+
         this.route.queryParams.subscribe((queryParams: Params) => {
             this.queryParams = queryParams;
         });
+
+        this.itemsService.items$
```

Comment:
The first, we don't have to subscribe on items$ observable. The second, it will be much better if we will receive watch object in route resolver.